bpo-31904 : Add support for VxWorks RTOS

kuhlenough opened the pull request:
This pull request enables cpython to cross-build on **VxWorks**, it is **obsolete** and only provided for reference of the scope of changes..  
Smaller more manageable PRs against the current master  as requested on python-dev will follow.

kuhlenough committed:
patch configure for VxWorks

kuhlenough committed:
bpo-31904 : Python should support VxWorks RTOS

All doesn't include shared if static build
Hardwire asscii encoding for VxWorks
default to UTF-8 (like Android)
Semicolon as path separator
If build is configure'd for static, propigate to Makefile
Add detection for oddly named VxWorks OPENSSL and HASH libraries
Fix some warnings with unused functions and macro re-def
Extra include in faulthandler.c
Avoid duplicate macro definition
Setup.py complies all VxWorks compatible modules
Update some extension modules to compile on VxWorks
VxWorks does not have "timezone" and daylight "constants"
Limited signal fields in VxWorks
undef DATE TIME for VxWorks

kuhlenough committed:
Any version of python should be OK for cross build.

kuhlenough committed:
undef DATE TIME for VxWorks

kuhlenough committed:
limited signal feilds in VxWorks

kuhlenough committed:
What's new file for bpo-30914

kuhlenough commented:
**Travis** failures appear completely unrelated to this pull request?

kuhlenough committed:
**install** and **commoninstall** targets do not include shared libraries if configured for static build
( i.e. ./configure --enable-shared=no )

kuhlenough committed:
revert Makefile.pre.in static build changes
(needs additional automake support)

kuhlenough committed:
Merge branch 'master' of https://github.com/kuhlenough/cpython

kuhlenough committed:
EOL whitespace removal

merwok commented on setup.py:
```diff
@@ -516,7 +524,7 @@ def detect_modules(self):
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
-        if cross_compiling:
+        if ( cross_compiling and not host_platform == 'vxworks'):
```
The extra space after the paren here is probably the source of the patchcheck warning on TravisCI.

kuhlenough committed:
Remove space

kuhlenough committed:
revert change

serhiy-storchaka commented on Modules/mmapmodule.c:
```diff
@@ -1463,9 +1463,9 @@ PyInit_mmap(void)
 
     setint(dict, "PAGESIZE", (long)my_getpagesize());
-
+#ifndef __VXWORKS__
```
I suggest to define my_getallocationgranularity to my_getpagesize above like on Unix.

kuhlenough replied:
Your correct this is not required.

serhiy-storchaka commented on Modules/posixmodule.c:
```diff
@@ -174,6 +174,7 @@ corresponding Unix manual entries for more information on calls.");
 #define HAVE_FSYNC      1
 #define fsync _commit
 #else
+#ifndef __VXWORKS__
```
Are all of this function absent? It looks suspicious that all of them are grouped alphabetically.

kuhlenough replied:
A couple are present in some configurations of VxWorks, but following comment is dated.
"Unix functions that the configure script doesn't check for"
The automake configure script finds all these without issue. So a better change would be to simply remove all these and let automake do it's job. I'm just not prepared to regression test the 1/2 dozen other OS referenced in this section.

serhiy-storchaka commented on Modules/socketmodule.c:
```diff
@@ -181,6 +181,12 @@ if_indextoname(index) -- return the corresponding interface name\n\
 
+#ifdef __VXWORKS__
+# include <ipcom_sock2.h>
+# define gethostbyaddr_r( a1, a2, a3, a4, a5, a6, a7 )  ipcom_gethostbyaddr_r( a1, a2, a3, a4, a5, a6, a7 )
```
Too long line. Why not just define gethostbyaddr_r to ipcom_gethostbyaddr_r?
```c
# define gethostbyaddr_r ipcom_gethostbyaddr_r
```

serhiy-storchaka commented on Modules/socketmodule.c:
```diff
@@ -532,7 +538,7 @@ set_error(void)
 }
 
-
+#ifndef __VXWORKS__
```
Is HAVE_HSTRERROR defined?

kuhlenough replied:
yes, but VxWorks has no h_errno also

serhiy-storchaka commented on setup.py:
```diff
@@ -851,20 +871,31 @@ def detect_modules(self):
             if krb5_h:
                 ssl_incs += krb5_h
-        ssl_libs = find_library_file(self.compiler, 'ssl',lib_dirs,
+        if host_platform == 'vxworks':
```
The code below is mostly duplicated. The only differences are few arguments. Wouldn't be easy to just set platform depending local variables for these arguments? The value for `libraries` also is used below.

kuhlenough committed:
Merge branch 'master' of https://github.com/kuhlenough/cpython

kuhlenough committed:
whitespace
(for Travis checkpatch)

kuhlenough committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	Modules/faulthandler.c
	Modules/mmapmodule.c

kuhlenough committed:
pull review changes

kuhlenough committed:
Merge branch 'master' of https://github.com/kuhlenough/cpython

Conflicts:
	Modules/mmapmodule.c

kuhlenough committed:
VxWorks has no h_errno reworked

kuhlenough committed:
Ignor readelf cross-compile error with VxWorks

kuhlenough committed:
ssl detections differences with varibles

serhiy-storchaka commented on Modules/socketmodule.c:
```diff
@@ -525,7 +532,7 @@ set_error(void)
 }
 
-
+/*#ifndef __VXWORKS__ */
```
If this no longer needed, remove it.

serhiy-storchaka commented on Modules/socketmodule.c:
```diff
@@ -5419,6 +5426,7 @@ Return the service name from a port number and protocol name.\n\
 The optional protocol name, if given, should be 'tcp' or 'udp',\n\
 otherwise any protocol will match.");
 
+#ifndef __VXWORKS__
```
Does an autoconf check work for getprotobyname?

kuhlenough replied:
No

serhiy-storchaka commented on Modules/timemodule.c:
```diff
@@ -1421,13 +1421,15 @@ PyInit_timezone(PyObject *m) {
     PyObject *otz0, *otz1;
     tzset();
+#ifndef __VXWORKS__
     PyModule_AddIntConstant(m, "timezone", timezone);
```
Does `timezone` not exist? What about `_timezone`?

kuhlenough replied:
No timezone, no _timezone,  you have to use tzset/tzget

serhiy-storchaka commented on configure:
```diff
@@ -2926,7 +2926,7 @@ $as_echo_n "checking for python interpreter for cross build... " >&6; }
 	    if $interp -c "import sys;sys.exit(not '.'.join(str(n) for n in sys.version_info[:2]) == '$PACKAGE_VERSION')"; then
 	        break
 	    fi
-            interp=
```
Why this line is removed? Is ` configure` generated from ` configure.ac`? What version of autoconf you have used?

kuhlenough replied:
I haven't checked in a regenerated configure yet. What version do you use?

serhiy-storchaka replied:
The last stable version is 2.69.

serhiy-storchaka commented on setup.py:
```diff
@@ -503,7 +511,7 @@ def add_gcc_paths(self):
     def detect_math_libs(self):
         # Check for MacOS X, which doesn't need libm.a at all
-        if host_platform == 'darwin':
+        if (host_platform == 'darwin' or host_platform == 'vxworks'):
```
Parenthesis are redundant.

serhiy-storchaka commented on setup.py:
```diff
@@ -516,7 +524,7 @@ def detect_modules(self):
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
-        if cross_compiling:
+        if (cross_compiling and not host_platform == 'vxworks'):
```
Parenthesis are redundant.

kuhlenough committed:
fix indent with patchcheck

kuhlenough committed:
add VxWorks support for os_cpu_count, spawnv, spawnve

kuhlenough committed:
Merge branch 'master' of https://github.com/kuhlenough/cpython

serhiy-storchaka commented on configure.ac:
```diff
@@ -418,9 +421,12 @@ if test "$cross_compiling" = yes; then
-	*-*-cygwin*)
-		_host_cpu=
-		;;
+        *-*-cygwin*)
```
Seems this is not indented as above lines.

kuhlenough committed:
Don't build or install shared libraries if
HAVE_DYNAMIC_LOADING is not defined

kuhlenough committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	setup.py

kuhlenough committed:
Restore mathlib() so VxWorks build doesn't fail looking for libm

kuhlenough committed:
Reverse ifeq HAVE_DYNAMIC_LOADING logic in Makefile.pre.in
so the _parse_makefile in sysconfig.py propigates the correct
settings to Setup.py

kuhlenough committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	setup.py

kuhlenough committed:
fix review issues

kuhlenough committed:
regenerated clinic/posixmodule.c.h
with correct checksum

kuhlenough committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	Modules/clinic/posixmodule.c.h

Commit:
test

Commit:
record changes before I remove test code

Commit:
wut

Commit:
compiles

Commit:
remove test files

Commit:
remove unessecary files

Commit:
remove files

Commit:
fix setup.py

Commit:
add PyList to char** code

Commit:
successfully exposed rtpSpawn

Commit:
fixstuff

Commit:
fix conflicts

Commit:
configure changed

Commit:
not sure

pablogsal committed:
Fix static linking extensions

pablogsal committed:
Add test for building extensions as builtins

pablogsal committed:
Add more extensions to the test

pablogsal committed:
Fix Windows Build

Commit:
Fix tests

Commit:
stage before merging bpo32232

Commit:
fix merge conflicts

Commit:
Merge branch 'pablogsal-bpo32232' into oscar

kuhlenough committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	setup.py

Commit:
refix os.pipe

Commit:
temp

Commit:
Merge branch 'master' of https://github.com/python/cpython into oscar

Commit:
fix random things

Commit:
fixed root of bug

Commit:
add random support for vxworks

Commit:
wat

Commit:
fix leaking fds"

ned-deily committed:
bpo-32726: Build macOS 10.9+ installer with private copy of Tcl/Tk 8.6

ned-deily committed:
Update NEWS, docs, and patchlevel for 3.7.0b1

matrixise committed:
bpo-32727: smtplib's SMTP.send_message behaves differently with from_addr and to_addrs (#5451)

Do not pass the name field in the 'from' address in the SMTP envelope.

bmwiedemann committed:
bpo-30693: zip+tarfile: sort directory listing (#2263)

tarfile and zipfile now sort directory listing to generate tar and zip archives
in a more reproducible way.

See also https://reproducible-builds.org/docs/stable-inputs/ on that topic.

zhangyangyu committed:
bpo-32583: Fix possible crashing in builtin Unicode decoders (#5325)

When using customized decode error handlers, it is possible for builtin decoders
to write out-of-bounds and then crash.

zhangyangyu committed:
Make site documentation more clear (#5461)

Mention only ImportError caused by importing sitecustomize.py/usercustomize.py
themselves will be silently ignored.

Commit:
need to remove debug messages later

csabella committed:
bpo-32735: Fix typo in f-strings datetime format specifier example (GH-5464)

The f-string example for using datetime format specifier does not match the given output.
Changed the format from %b to %B so it matches the output of "January".

ned-deily committed:
finished 3.7.0b1 - on to b1+

ned-deily committed:
Update Doc build for split off of 3.7 branch

miss-islington committed:
bpo-32565: Add missed versionadded directives for all new opcodes. (GH-5199) (GH-5470)

(cherry picked from commit 12e7cd8a51956a5ce373aac692ae6366c5f86584)

Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

miss-islington committed:
[3.7] bpo-32711: Fix warnings for Python/ast_unparse.c (GH-5426) (#5475)

* bpo-32711: Fix warnings for Python/ast_unparse.c
(cherry picked from commit 83ab995871ffd504ac229bdbf5b9e9ffc1032815)

Co-authored-by: Stéphane Wirtel <stephane@wirtel.be>

miss-islington committed:
Implement TimerHandle.when() (GH-5473) (GH-5480)

(cherry picked from commit 3d4dbd8f019c0bbac99fc9248077044ff1039ca3)

Co-authored-by: Andrew Svetlov <andrew.svetlov@gmail.com>

miss-islington committed:
Fix typo in hamt.c comments (GH-5478) (#5484)

(cherry picked from commit 01a0cb891694cf73b86f799c48d8c78de1b8f74c)

Co-authored-by: Dmitry Alimov <dvalimov@gmail.com>

miss-islington committed:
bpo-32436: Fix compiler warning (GH-5483) (GH-5486)

(cherry picked from commit 55e0839f2672e029c2b96514028c77c31ffbe41f)

Co-authored-by: Yury Selivanov <yury@magic.io>

miss-islington committed:
bpo-32674: Improve the docstring for __import__ (GH-5339) (GH-5488)

Clarify that the level argument is used to determine whether to
perform absolute or relative imports: 0 is absolute, while a positive number
is the number of parent directories to search relative to the current module.
(cherry picked from commit 461d225b195eec5269f317323b41115516144c41)

Co-authored-by: oldk <oldk1331@users.noreply.github.com>

miss-islington committed:
bpo-32733: Make test_coroutines robust against -Werror (GH-5487) (GH-5490)

(cherry picked from commit 2efb9735899c7c642f3e75a121af82b1679577d3)

Co-authored-by: Nathaniel J. Smith <njs@pobox.com>

miss-islington committed:
Revert "bpo-31356: Add context manager to temporarily disable GC GH-5495 (#5496)

This reverts commit 72a0d218dcc94a3cc409a9ef32dfcd5a7bbcb43c.

The reverted commit had a few issues so it was unanimously decided
to undo it. See the bpo issue for details.
(cherry picked from commit 383b32fe108ea627699cc9c644fba5f8bae95d73)

Co-authored-by: Yury Selivanov <yury@magic.io>

Commit:
test

miss-islington committed:
bpo-32614: Modify re examples to use a raw string to prevent warning (GH-5265) (#5499)

Modify RE examples in documentation to use raw strings to prevent DeprecationWarning.
Add text to REGEX HOWTO to highlight the deprecation.  Approved by Serhiy Storchaka.
(cherry picked from commit 66771422d0541289d0b1287bc3c28e8b5609f6b4)

Co-authored-by: Cheryl Sabella <cheryl.sabella@gmail.com>

miss-islington committed:
bpo-32734: Fix asyncio.Lock multiple acquire safety issue (GH-5466) (#5501)

(cherry picked from commit 2f79c014931cbb23b08a7d16c534a3cc9607ae14)

Co-authored-by: Bar Harel <bzvi7919@gmail.com>

warsaw committed:
bpo-32303 - Consistency fixes for namespace loaders (GH-5481) (#5503)

* Make sure ``__spec__.loader`` matches ``__loader__`` for namespace packages.
* Make sure ``__spec__.origin` matches ``__file__`` for namespace packages.

https://bugs.python.org/issue32303
https://bugs.python.org/issue32305
(cherry picked from commit bbbcf8693b876daae4469765aa62f8924f39a7d2)

Co-authored-by: Barry Warsaw <barry@python.org>

miss-islington committed:
bpo-32604: Fix memory leaks in the new _xxsubinterpreters module. (GH-5507)

(cherry picked from commit 4e9da0d163731caa79811c723c703ee416c31826)

Co-authored-by: Eric Snow <ericsnowcurrently@gmail.com>

miss-islington committed:
bpo-32739: Show default value for rotate() (GH-5485) (GH-5514)

(cherry picked from commit 589c718a8e3bde017350f248f7f1c009240eb52b)

Co-authored-by: Raymond Hettinger <rhettinger@users.noreply.github.com>

miss-islington committed:
[3.7] bpo-32746: Fix multiple typos (GH-5144) (GH-5520)

Fix typos found by codespell in docs, docstrings, and comments.
(cherry picked from commit c3d9508ff22ece9a96892b628dd5813e2fb0cd80)

Co-authored-by: Leo Arias <leo.arias@canonical.com>

miss-islington committed:
bpo-32720: Fixed the definition for width and precision in format mini-language doc (GH-5482) (GH-5524)

Changed the definition of width and precision from "integer" to "digit+" in format mini-language doc.
(cherry picked from commit 8b5fa289fdb04b6b919cf95fa99246aa872e47a8)

Co-authored-by: nathankerr96 <nathankerr96@gmail.com>

miss-islington committed:
[3.7] bpo-32691: Use mod_spec.parent when running modules with pdb (GH-5510)

Previously the module name was used, which broke relative imports when pdb was run against a plain module or submodule.
(cherry picked from commit 38bfa8418f5d39bcc7478b8f7aef4a632c26172e)

Co-authored-by: Mario Corchero <mariocj89@gmail.com>

miss-islington committed:
Use assertTrue() instead of deprecated assert_(). (GH-5526) (GH-5527)

(cherry picked from commit 95c3262762f7d0da97ad633f40bed7cd2df9814c)

Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

miss-islington committed:
Fix typo -- missing "not" (GH-5528) (GH-5530)

miss-islington committed:
bpo-32765: Update configdialog General tab create page docstring (GH-5529) (GH-5537)

Add new entries to the widget list.
(cherry picked from commit 845d86485e35a26478aedb3dc127d632fdc65759)

Co-authored-by: Cheryl Sabella <cheryl.sabella@gmail.com>

miss-islington committed:
bpo-30928: IDLE - update NEWS.txt. (GH-5539) (GH-5540)

Add entry for uniform lower context; add 'since' to be explicit. 
(cherry picked from commit 05e806767b857b1eab838e712828e3a7d57cabf1)

Co-authored-by: Terry Jan Reedy <tjreedy@udel.edu>

miss-islington committed:
[3.7] bpo-8722: Document __getattr__ behavior with AttributeError in property (GH-5543)

When `__getattr__` is implemented, attribute lookup will always fall back to that,
even if the initial failure comes from `__getattribute__` or a descriptor's `__get__`
method (including property methods).
(cherry picked from commit d1f318105b8781b01f3507d5cb0fd841b977d5f2)

Co-authored-by: Cheryl Sabella <cheryl.sabella@gmail.com>

miss-islington committed:
bpo-32777: Fix _Py_set_inheritable async-safety in subprocess (GH-5560) (GH-5562)

Fix a rare but potential pre-exec child process deadlock in subprocess on POSIX systems when marking file descriptors inheritable on exec in the child process.  This bug appears to have been introduced in 3.4 with the inheritable file descriptors support.

This also changes Python/fileutils.c `set_inheritable` to use the "slow" two `fcntl` syscall path instead of the "fast" single `ioctl` syscall path when asked to be async signal safe (by way of being asked not to raise exceptions).  `ioctl` is not a POSIX async-signal-safe approved function.

ref: http://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html
(cherry picked from commit c1e46e94de38a92f98736af9a42d89c3975a9919)

Co-authored-by: Alexey Izbyshev <izbyshev@users.noreply.github.com>

Commit:
add timeout

miss-islington committed:
bpo-30693: Fix tarfile test cleanup on MSWindows (GH-5557) (GH-5567)

it was using our mocked listdir to check when the files were gone.
(cherry picked from commit 4ad703b7ca463d1183539277dde90ffb1c808487)

Co-authored-by: Bernhard M. Wiedemann <githubbmw@lsmod.de>

miss-islington committed:
bpo-6135: Fix subprocess.check_output doc to mention changes in 3.6 (GH-5564) (GH-5572)

Fixes the documentation for `subprocess.check_output()` not mentioning that the encoding and errors parameters were added in 3.6.
(cherry picked from commit fc1ce810f1da593648b4d19e7d582a235ec1dd37)

Co-authored-by: Brice Gros <brice-gros@users.noreply.github.com>

miss-islington committed:
Fix typo in whatsnew/3.7.rst (GH-5551) (GH-5553)

now longer -> no longer
(cherry picked from commit c309bcfb9fb295e70a235c461d9edcaa54c821d0)

Co-authored-by: Dag Heyman <dag@yubico.com>

miss-islington committed:
bpo-32720: Fixed the replacement field grammar documentation. (GH-5544) (GH-5546)

`arg_name` and `element_index` are defined as `digit`+ instead of `integer`.
(cherry picked from commit 7a561afd2c79f63a6008843b83733911d07f0119)

Co-authored-by: Mariatta <Mariatta@users.noreply.github.com>

Commit:
test

Commit:
change char to wchar

miss-islington committed:
bpo-32802: Fix Travis build (GH-5589) (GH-5590)

Fix bug in travis configuration where it did not run the tests when
a change includes both code and doc changes.
(cherry picked from commit 32921f90821ab54ffb757b7e996e5b7a71fac25e)

Co-authored-by: Stéphane Wirtel <stephane@wirtel.be>

Commit:
fix problem with sysroot addition;

Commit:
Merge branch 'master' into oscar

Commit:
add vxwapi as shared lib

miss-islington committed:
bpo-32585: Add tkinter.ttk.Spinbox. (GH-5221) (GH-5592)

(cherry picked from commit a48e78a0b7761dd74f1d03fc69e0f6caa6f02fe6)

Co-authored-by: Alan D Moore <me@alandmoore.com>

miss-islington committed:
bpo-32775: Fix regular expression warnings in fnmatch. (GH-5583) (GH-5596)

fnmatch.translate() no longer produces patterns which contain set
operations.

Sets starting with '[' or containing '--', '&&', '~~' or '||' will
be interpreted differently in regular expressions in future versions.
Currently they emit warnings. fnmatch.translate() now avoids producing
patterns containing such sets by accident.
(cherry picked from commit 23cdbfa744f0ec0e9e7575d378df4cb758691cd3)

Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

miss-islington committed:
Make formatting of some return codes conforming to the general style. (GH-5587) (GH-5597)

(cherry picked from commit 5bb0005f9ff768ac443924b4bb26c3818ce8dc5a)

Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

miss-islington committed:
Fix some warnings produced by different compilers. (GH-5593) (GH-5600)

(cherry picked from commit bfe4fd5f2e96e72eecb5b8a0c7df0ac1689f3b7e)

Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

Commit:
Shared Libs finalyl works! site.py had a abspath function that needed to be changed

miss-islington committed:
bpo-30157: Fix csv.Sniffer.sniff() regex pattern. (GH-5601) (GH-5602)

Co-authored-by: Jake Davis <jcdavis@awedge.net>
(cherry picked from commit 2411292ba8155327125d8a1da8a4c9fa003d5909)

miss-islington committed:
travis: Use -O3 option (GH-5599)

We don't use debugger on Travis.
(cherry picked from commit 8ff53564730f7ba503f5ad94418c309c48e8516d)

Co-authored-by: INADA Naoki <methane@users.noreply.github.com>

miss-islington committed:
bpo-32800: Update link to w3c doc for xml default namespaces (GH-5609) (GH-5611)

The new link is given in a red box on the old page.
(cherry picked from commit 8d1f2f40389c476d9240ec87a33da0373597c6af)

Co-authored-by: sblondon <sblondon@users.noreply.github.com>

miss-islington committed:
bpo-32792: Preserve mapping order in ChainMap() (GH-5586) (#GH-5617)

(cherry picked from commit 3793f95f98c3112ce447288a5bf9899eb9e35423)

Co-authored-by: Raymond Hettinger <rhettinger@users.noreply.github.com>

miss-islington committed:
bpo-11015: Update test.support documentation (GH-5619)

(cherry picked from commit 988fb28431d3a3fc5dc6eb657c8a4baacc04d9ce)

Co-authored-by: Cheryl Sabella <cheryl.sabella@gmail.com>

miss-islington committed:
bpo-32604: Make _xxsubinterpreters build on Windows (GH-5624)

This is not the ideal solution; this means that a test module is now
always included in the main python3x.dll.  However, we're already
including xxsubtype, so why not?
(cherry picked from commit 310b05289b5d9550040f469e60b5e8e77f1022b6)

Co-authored-by: Zachary Ware <zachary.ware@gmail.com>

zware committed:
[3.7] Add short-circuit for doc changes to AppVeyor (GH-5626)

(cherry picked from commit 28607e0dd9417ce44a109980ffd60697c1afdea0)

miss-islington committed:
Fix AppVeyor doc short-circuit (GH-5633)

(cherry picked from commit 6ea20fc719dcaf102d2cefa1786f0421514f7a58)

miss-islington committed:
bpo-32815: Improve docs on the subprocess API *text* parameter (GH-5622) (GH-5631)

Describe *text* as an alias for *universal_newlines* in more places that people are likely to be referred to.
(cherry picked from commit e14c01037877768a3fe766e50d14993bd5d8a67e)

Co-authored-by: Pablo Galindo <Pablogsal@gmail.com>

miss-islington committed:
Fix typo in Include/objimpl.h, the word "has" was missing (GH-5568) (GH-5569)


It now reads: ...be aware that Python has no control over...
(cherry picked from commit 517da1e58f4c489d4b31579852cde5f7113da08e)

Co-authored-by: Alexey <forestbiiird@gmail.com>

Commit:
remove debug messages; add def for gethostbyaddr

Commit:
remove broken ifdef

Commit:
fix static builds

miss-islington committed:
bpo-32221: makeipaddr(): remove interface part + speedup (GH-5449) (GH-5449) (#5641)

(cherry picked from commit 7766b96ab80b04509bbac708ee5ecf3c1c5934fc)

Co-authored-by: Коренберг Марк <socketpair@gmail.com>

miss-islington committed:
bpo-32826: Add "encoding=utf-8" to open() in idle_test/test_help_about. (GH-5639)


GUI test test_file_buttons() only looks at initial ascii-only lines,
but failed on systems where open() defaults to 'ascii' because
readline() internally reads and decodes far enough ahead to encounter
a non-ascii character in CREDITS.txt.
(cherry picked from commit f34e03ec0ea6a4cef8d966087c77e616c4a5893b)

Co-authored-by: Terry Jan Reedy <tjreedy@udel.edu>

miss-islington committed:
bpo-29248: Fix os.readlink() on Windows (GH-5577)


The PrintNameOffset field of the reparse data buffer
was treated as a number of characters instead of bytes.
(cherry picked from commit 3c34aad4e7a95913ec7db8e5e948a8fc69047bf7)

Co-authored-by: SSE4 <tomskside@gmail.com>

miss-islington committed:
bpo-32837: IDLE - require encoding argument for textview.view_file. (GH-5646)


Using the system and place-dependent default encoding for open()
is a bad idea for IDLE's system and location-independent files.
(cherry picked from commit 688722cedd6437910ff185ecf94fb3b749ad37f2)

Co-authored-by: Terry Jan Reedy <tjreedy@udel.edu>

miss-islington committed:
bpo-32827: Fix usage of _PyUnicodeWriter_Prepare() in decoding errors handler. (GH-5636) (GH-5650)

(cherry picked from commit b7e2d67f7c035f09c921ca4e7a36529cd502ccf7)


Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

miss-islington committed:
bpo-30579: Docs for dynamic traceback creation (GH-5653)

(cherry picked from commit aec7532ed3ccbd29d3429a3f375e25f956c44003)

Co-authored-by: Nick Coghlan <ncoghlan@gmail.com>

miss-islington committed:
bpo-32370: Use the correct encoding for ipconfig output in the uuid module. (GH-5608)

(cherry picked from commit da6c3da6c33c6bf794f741e348b9c6d86cc43ec5)

Co-authored-by: Segev Finer <segev208@gmail.com>

miss-islington committed:
bpo-31787: Prevent refleaks when calling __init__() more than once (GH-3995)

(cherry picked from commit d019bc8319ea35e93bf4baa38098ff1b57cd3ee5)

Co-authored-by: Oren Milman <orenmn@gmail.com>

miss-islington committed:
bpo-31787: Skip refleak check when _hashlib is not available (GH-5660)

(cherry picked from commit f0bc645dfede8118c84844bad319cd952c4d1905)

Co-authored-by: INADA Naoki <methane@users.noreply.github.com>

miss-islington committed:
bpo-27846: Delete incorrect note in base64 docs (GH-5666)

This note incorrectly stated that "Base64 has an expansion factor of 6
to 4" (it is actually 4 to 3). It was decided to remove the note.

(cherry picked from commit 88c38a4049ded0e1f197fec3e76be7c0c0e83d7a)

Co-authored-by: Zackery Spytz <zspytz@gmail.com>

miss-islington committed:
remove mercurial dot files (GH-5558)

(cherry picked from commit 2b86f4cb2cd339f545804a9abb8c4d4cc28d9488)

Co-authored-by: Benjamin Peterson <benjamin@python.org>

miss-islington committed:
Fix installation instructions for *nix (GH-5605)


Remove pkg_add -r python from FreeBSD installation section.
Moved to OpenBSD.
(cherry picked from commit 3384d38d51a2c3450e742175db5d6d638fa5d2eb)

Co-authored-by: Eitan Adler <grimreaper@users.noreply.github.com>

miss-islington committed:
bpo-32841: Fix cancellation in awaiting asyncio.Condition (GH-5665)

(cherry picked from commit 5746510b7aef423fa4afc92b2abb919307b1dbb9)

Co-authored-by: Bar Harel <bzvi7919@gmail.com>

Commit:
actually fix wrong module builds whne converting to use _vxworks instea do findivitual checks

miss-islington committed:
Update comment in posixmodule.c (GH-5681)


A closing parentheses was missing.

Signed-off-by: Ngie Cooper <yaneurabeya@gmail.com>
(cherry picked from commit 7745ec4e356ac1f4eaf43b155f4482c20a907d48)

Co-authored-by: ngie-eign <1574099+ngie-eign@users.noreply.github.com>

Commit:
link the proper alloca for ctypes

miss-islington committed:
Update and sync importlib.resources documentation (GH-5694) (#5697)

(cherry picked from commit 997b8c140e768c849fa99d172245a7688af20a89)

Co-authored-by: Barry Warsaw <barry@python.org>

miss-islington committed:
bpo-32436: Add docs for contextvars (GH-5685)

(cherry picked from commit bd093355a6aaf2f4ca3ed153e195da57870a55eb)

Co-authored-by: Yury Selivanov <yury@magic.io>

Commit:
add proper include flags for openssl build

miss-islington committed:
Correct the code example in Python 3.7's What's New (GH-5696)


There was an extra dash in the example for re.sub().
(cherry picked from commit b65cb163d6e125606070c54dc757f421fe419b59)

Co-authored-by: xpvpc <32843902+xpvpc@users.noreply.github.com>

Commit:
VxWorks doesnt support O_NONBLOCK Flag

miss-islington committed:
bpo-32604: Clean up created subinterpreters before runtime finalization. (gh-5710)

(cherry picked from commit 4c6955e2b0ccf88c705f8d1fac685a8e65f9699e)

Co-authored-by: Eric Snow <ericsnowcurrently@gmail.com>

miss-islington committed:
bpo-32860: Fix a missing asterisk in the documentation for glob.iglob() (GH-5712)

(cherry picked from commit 6887d86e9a04b9c5718b08e5d9e1ca3cc1f53721)

Co-authored-by: Zackery Spytz <zspytz@gmail.com>

miss-islington committed:
DOC: fix documentation for copyright and credits (GH-5706)


Adapt documentation for `copyright` and `credits` to reality.  Previously, the documentation implied that all each of `copyright`,
`credits`, and `license`, would print a message to call the object in order to see the full text.  In reality, only `license` exhibits this
behaviour, and `copyright` and `credit` print their full text either when printed, displayed, or called.
(cherry picked from commit 243d6d71265be1996a8326002deae429106a5fba)

Co-authored-by: Gerrit Holl <gerrit.holl@gmail.com>

miss-islington committed:
bpo-32852: Fix trace changing sys.argv to tuple. (GH-5692)

(cherry picked from commit 9f4223261fd129ad7b9a09b2b0d625d1bb90b22b)

Co-authored-by: Kyle Altendorf <sda@fstab.net>

miss-islington committed:
Fix a typo in asyncio docs (GH-5721)

(cherry picked from commit 17ab8f0e8e2b2f66d11fbb95aa9936544a5e1a9a)

Co-authored-by: Andrew Svetlov <andrew.svetlov@gmail.com>

miss-islington committed:
bpo-30638: Add clinic to `make regen-all` (GH-5671)


Also, use PYTHON_FOR_REGEN for clinic and blake2s_impl.c rather than PYTHON_FOR_BUILD, and update .travis.yml to make use of the change.
(cherry picked from commit d6ff8a7037903497eff95fa32bdac2b6adf71505)

Co-authored-by: Zachary Ware <zachary.ware@gmail.com>

miss-islington committed:
[3.7] Improves the ability to build in CI (GH-5729)

miss-islington committed:
Clean up Travis config (GH-5727)

(cherry picked from commit 7eb3f8226ea7b79dae4e4e8b05730cfe0d9af7c1)

Co-authored-by: Zachary Ware <zachary.ware@gmail.com>

ilevkivskyi committed:
bpo-31333: Re-implement ABCMeta in C (GH-5733)

This adds C versions of methods used by ABCMeta that
improve performance of various ABC operations.

ilevkivskyi committed:
bpo-25988: Emit a warning when use or import ABCs from 'collections'. (GH-5734)

ned-deily committed:
bpo-25988: add NEWS entry for 3.7.0b2 (#5743)

miss-islington committed:
bpo-32869: Fix incorrect dst buffer size for MultiByteToWideChar (GH-5739)


This function expects the destination buffer size to be given
in wide characters, not bytes.
(cherry picked from commit b3b4a9d3001f1fc7df8efcccdce081de54fa5eab)

Co-authored-by: Alexey Izbyshev <izbyshev@users.noreply.github.com>

miss-islington committed:
bpo-31333: Fix typo in whatsnew/3.7.rst (GH-5744)

(cherry picked from commit 3fb813d2c67fe28cc98ae51e53a6890294b6e423)

Co-authored-by: Terry Jan Reedy <tjreedy@udel.edu>

miss-islington committed:
bpo-31972: Improve docstrings for pathlib classes (GH-5310)

(cherry picked from commit dfa015cf77a148d229ddc0a5b063562c9f9664f2)

Co-authored-by: chason <chason@gmail.com>

miss-islington committed:
Improve error message for "setup.py upload" without dist files (GH-21060)

(cherry picked from commit 08a6926b2584040fe3c3f06263b0b5f1fbbdc24c)

Co-authored-by: Éric Araujo <merwok@netwok.org>

miss-islington committed:
bpo-32682: Improve libz version parsing in test_zilb (GH-5347)

(cherry picked from commit 4c7108a77144493d0aa6fc0105b67d3797e143f5)

Co-authored-by: pmp-p <pmp-p@users.noreply.github.com>

miss-islington committed:
bpo-30121: Fix test_subprocess for Windows Debug builds (GH-5758)

(cherry picked from commit 5537646bfacec463b450871dde31cb06c44a0556)

Co-authored-by: Zachary Ware <zachary.ware@gmail.com>

miss-islington committed:
bpo-32409: Ensures activate.bat can handle Unicode contents (GH-5765)

(cherry picked from commit 6240917b773b52f8883387b9e3a5f327a4372068)

Co-authored-by: Steve Dower <steve.dower@microsoft.com>

miss-islington committed:
closes bpo-32859: Don't retry dup3() if it is not available at runtime (GH-5708)

os.dup2() tests for dup3() system call availability at runtime,
but doesn't remember the result across calls, repeating
the test on each call with inheritable=False.

Since the caller of os.dup2() is expected to hold the GIL,
fix this by making the variable holding the test result static.
(cherry picked from commit b3caf388a0418f6c031e4dbdcc0c1ce7e5cc36bd)

Co-authored-by: Alexey Izbyshev <izbyshev@users.noreply.github.com>

miss-islington committed:
bpo-31848: Fix broken error handling in Aifc_read.initfp() when the SSND chunk is not found (GH-5240)


Initialize self._ssnd_chunk so that aifc.Error is raised as intended,
not AttributeError.
(cherry picked from commit 80d20b918bd8a882043c493a7f958333ecb41727)

Co-authored-by: Zackery Spytz <zspytz@gmail.com>

miss-islington committed:
bpo-32604: Swap threads only if the interpreter is different. (gh-5783)

The CPython runtime assumes that there is a one-to-one relationship (for a given interpreter) between PyThreadState and OS threads. Sending and receiving on a channel in the same interpreter was causing crashes because of this (specifically due to a check in PyThreadState_Swap()). The solution is to not switch threads if the interpreter is the same.
(cherry picked from commit f53d9f2778a87bdd48eb9030f782a4ebf9e7622f)

Co-authored-by: Eric Snow <ericsnowcurrently@gmail.com>

miss-islington committed:
bpo-32540: Update venv documentation (GH-5736)

If the directory already exists, it will be re-used.

(cherry picked from commit e8eb972514cad6086b752754c1d34d703c04dd82)

Co-authored-by: TROUVERIE Joachim <joachim.trouverie@linoame.fr>

miss-islington committed:
bpo-28886: doc: Move deprecated abc decorators to separate section (GH-176)

(cherry picked from commit 52c6b89796a7ec391db20281e05b256f57e97b35)

Co-authored-by: Harshul jain <harshulrobo@gmail.com>

miss-islington committed:
bpo-32008: don't use PROTOCOL_TLSv1 in example (GH-5789)


It's bad form to pin to an old version of TLS. ssl.SSLContext has the right
protocol default, so let's not pass anyway.
(cherry picked from commit e9edee0b65650c4f9db90cefc2e9a8125bad762c)

Co-authored-by: Benjamin Peterson <benjamin@python.org>

miss-islington committed:
bpo-32889: update valgrind suppressions (GH-5779)


Py_ADDRESS_IN_RANGE was renamed address_in_range in 3.6
(commit 3924f93794fd740c547b44884f73303196475cd5).
(cherry picked from commit ba518804bf4c1ea01df5e622b333d3116cbaa3bd)

Co-authored-by: Paul Price <price@astro.princeton.edu>

miss-islington committed:
bpo-32874: IDLE: add tests for pyparse (GH-5755)


There are no code changes other than comments and docstrings.
(cherry picked from commit c84cf6c03fce1fb73bfaf91d7909f1c2708f14a2)

Co-authored-by: Cheryl Sabella <cheryl.sabella@gmail.com>

miss-islington committed:
closes bpo-32898: Fix debug build crash with COUNT_ALLOCS (GH-5800)

(cherry picked from commit 745dc65b17b3936e3f9f4099f735f174d30c4e0c)

Co-authored-by: Eddie Elizondo <eduardo.elizondorueda@gmail.com>

miss-islington committed:
bpo-32905: IDLE - remove unused code in pyparse module (GH-5807)


dump is similar to print but less flexible.  lastopenbracketpos is now always initialized in _study2, as was stmt_bracketing, so the class settings are not needed.  get_last_open_bracket_pos is never called.
(cherry picked from commit 451d1edaf4d27c4e632d81246d308e8dd6ea945f)

Co-authored-by: Terry Jan Reedy <tjreedy@udel.edu>

miss-islington committed:
bpo-32556: nt._getfinalpathname, nt._getvolumepathname and nt._getdiskusage now correctly convert from bytes. (GH-5761)

(cherry picked from commit 23ad6d0d1a7a6145a01494f4f3913a63d1f0250c)

Co-authored-by: Steve Dower <steve.dower@microsoft.com>

miss-islington committed:
bpo-32457: Improves handling of denormalized executable path when launching Python (GH-5756)

(cherry picked from commit 48e8c82fc63d2ddcddce8aa637a892839b551619)

Co-authored-by: Steve Dower <steve.dower@microsoft.com>

Commit:
merge

Commit:
fix redef issue with NONE in _pickle

Commit:
increase subprocess size

Commit:
remove all debug messages

Commit:
fix broken else

Commit:
move subprocess pipe duplicate into more reasonable placeds and replace function calls with standard numbers

Commit:
fix bug with spawned subprocess not closing p2cwrite

Commit:
restore correct path after spawning subprocess

Commit:
switch default encoding to ascii

Commit:
fix hash randomization at initialization

Commit:
skip some test cases

Commit:
ensure correct openssl libs included

Commit:
forgot test

Commit:
keep fixing test cases

Commit:
lost a days worth of work cuz forgot to push before rebuild >.>, this time actually push

Commit:
skip osfork tests

Commit:
whoops on beginswith

Commit:
skip ssl tests if vxworks

Commit:
idk

Commit:
expose and use vxworks pathIsAbsolute

Commit:
realpath is currently broken

Commit:
fix random

Commit:
keep fixing test modules

Commit:
keep fixing path issues

Commit:
skip test

Commit:
fixed select blocking

Commit:
fix last commit

Commit:
yeah no select isnt fixed yet

kuhlenough committed:
Merge pull request #2 from kuhlenough/oscar

Merge development branch into master

kuhlenough committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	.github/appveyor.yml
#	.gitignore
#	.travis.yml
#	Doc/extending/newtypes.rst
#	Doc/library/contextvars.rst
#	Doc/library/dis.rst
#	Doc/library/test.rst
#	Doc/tools/templates/indexsidebar.html
#	Doc/whatsnew/3.7.rst
#	Include/patchlevel.h
#	Lib/_py_abc.py
#	Lib/idlelib/NEWS.txt
#	Lib/idlelib/idle_test/test_pyparse.py
#	Lib/idlelib/pyparse.py
#	Lib/test/test_abc.py
#	Lib/test/test_aifc.py
#	Lib/test/test_os.py
#	Mac/BuildScript/build-installer.py
#	Mac/BuildScript/resources/ReadMe.rtf
#	Mac/BuildScript/resources/Welcome.rtf
#	Misc/ACKS
#	Misc/NEWS.d/3.7.0b1.rst
#	Misc/NEWS.d/next/Library/2018-01-18-13-09-00.bpo-32585.qpeijr.rst
#	Modules/_abc.c
#	Modules/posixmodule.c
#	Python/importlib.h
#	Python/importlib_external.h
#	README.rst

ooosssososos committed:
remove debug print

ooosssososos committed:
Merge pull request #3 from kuhlenough/oscar

remove debug print

ooosssososos committed:
remove testing code

ooosssososos committed:
remove testing code

ooosssososos committed:
fixed the random problem in vxworks seeding

ooosssososos committed:
change modificitation to skip

ooosssososos committed:
remove #endif

ooosssososos committed:
Merge pull request #4 from kuhlenough/oscar

Remove test code

kuhlenough committed:
Revert "Update Doc build for split off of 3.7 branch"

This reverts commit 7f38637853c25a72abeea75f854d81a91b087cb5.

# Conflicts:
#	Doc/tools/templates/indexsidebar.html

kuhlenough committed:
Merge branch 'master' of https://github.com/kuhlenough/cpython

kuhlenough committed:
Revert "Fix static linking extensions"

This reverts commit 18463dd0a156d4dcfca9e2344c4cb7980a4780a5.

# Conflicts:
#	Include/pystate.h

kuhlenough committed:
Revert "Fix Windows Build"

This reverts commit 0ad214f51f28a21483128480955b2286ebf56500.

kuhlenough committed:
Revert "Add more extensions to the test"

This reverts commit dd55e39564c310890e745cdb7f39060ee3c11bc8.

# Conflicts:
#	Lib/test/test_support.py

kuhlenough committed:
revert linefeed changes

kuhlenough committed:
Revert "Fix static linking extensions"

This reverts commit 18463dd0a156d4dcfca9e2344c4cb7980a4780a5.

kuhlenough committed:
clean-up cruff

nascheme commented on Lib/multiprocessing/util.py:
```diff
@@ -410,6 +410,13 @@ def _flush_std_streams():
+        import subprocess
+        try:
+            return subprocess.Popen(args).pid
+        except:
```
This is not good.  First, 'try' is bare 'except' is bad news, hides bugs.  Second, you are doing the 'vxworks' test on every function call, for every platform.  Instead you could have something like:
```
def spawnv_passfds(...):
    ...

if 'vxworks' in sys.platform:
  # a comment explaining why this is needed
  def spawnv_passfds(...):
    # implementation of vxworks special version
```

nascheme commented on Lib/platform.py:
```diff
@@ -753,7 +753,7 @@ def _syscmd_uname(option, default=''):
     try:
         f = os.popen('uname %s 2> %s' % (option, DEV_NULL))
-    except (AttributeError, OSError):
+    except (AttributeError, OSError, ValueError):
```
Looks suspicious.  Does os.popen() now return different exceptions?  I suspect you should fix os.popen() to trap ValueError and return OSError instead.  Otherwise everyone who uses os.popen() has to deal with the new exception case.

nascheme commented on Lib/posixpath.py:
```diff
@@ -65,13 +69,17 @@ def isabs(s):
+                 # way to find if a path is absolute. V7COR-3074, F7233
+        if not isinstance(s, str):
+            s = s.decode(sys.getdefaultencoding())
+        return bool(_vxwapi.isAbs(s))
```
Again, you are doing the _vxworks test on every function call for every platform.  Not a huge deal but isabs() might be called a lot.  Instead, define a new isabs() on vxworks.

nascheme commented on Lib/posixpath.py:
```diff
@@ -401,8 +409,18 @@ def _joinrealpath(path, rest, seen):
+                ind = rest.find(':')
+                ind = 0 if ind < 0 else ind
+                path = rest[:ind]
+                rest = rest[ind:]
```
Again, runtime testing of _vxworks.  Better to define a special _joinrealpath() function at module import time.

nascheme commented on Lib/test/test_netrc.py:
```diff
@@ -138,7 +138,7 @@ def test_file_not_found_in_home(self):
                           file='unlikely_netrc')
-
+    @unittest.skipIf('vxworks' in sys.platform, 'VxWorks does not require or have a HOME directory')
     def test_home_not_set(self):
```
Sprinkling verbose tests like this throughout the code is not the way. Instead, you should define a global flag at the top of the module, named based on the feature that is not on VxWorks.  E.g.
```
_HAVE_HOME = 'vxworks' not in sys.platform
```
Then, for the test functions:
```
@unittest.skipIf(not _HAVE_HOME)
```
Better yet perhaps is to add those flags to a central module, e.g. `platform`.  Doing it this way makes it more clear what is going on.  Also, if a new platform is added, the tests for platform specific features becomes easier to understand.

nascheme commented on Modules/_multiprocessing/multiprocessing.h:
```diff
@@ -32,8 +32,12 @@
+#  endif
+#  ifndef FALSE
+#    define FALSE 0
+#  endif
```
Doesn't look correct to me.  multiprocessing requires TRUE = 1.  I think you should 'undef' the existing TRUE/FALSE and define what multiprocessing expects.
